cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/partitioned/ConsumablePartition.java

```diff
@@ -54,7 +54,7 @@
   void take();
 
   /**
-   * Marks the ProcessState as AVAILABLE, resets the timestamp to 0, and increments the number of failures by 1
+   * Marks the ProcessState as AVAILABLE and resets the timestamp to 0
```

Comment:
This changes the semantics of an exitsting API. Better keep the existing one and introduce a new one for retrying without incrementing the failure count.